lib/route.js

```diff
@@ -269,7 +269,7 @@ function buildRouting (options) {
       if (prefixing === false) {
         // run 'onRoute' hooks
         for (const hook of this[kHooks].onRoute) {
-          hook.call(this, opts)
+          hook.call(this, this, opts)
```

Comment:
Do not forget to update docs for this one

Reply:
I'll do the docs (except codemods) once we decide what we want at https://github.com/fastify/fastify/pull/5670#discussion_r1749147260